### PR TITLE
[Buster] Upgrade to 4.9.2

### PR DESCRIPTION
**Pull request by kay0u**
Waiting for 3.8

### CODE REVIEW

**Comment by maniackcrudelis** [comment]
I can run a test with package check if you want, I have VM on Buster for that

**Comment by kay0u** [comment]
That might be greate, but a test with 3.8 will also be necessary.

**Comment by maniackcrudelis** [comment]
Ok, I have also a VM on 3.8, I can run both tests.

**Comment by maniackcrudelis** [comment]
```
Configuring system user... [00h00m,03s]
Warning: yunohost.hook <lambda> - [2109.1] Invalid argument: --
```

File: scripts/_common.sh [hunk 1]
```diff
@@ -5,7 +5,10 @@
 #=================================================
 
 # dependencies used by the app
-pkg_dependencies="openssl php7.0-mbstring  php7.0-curl php7.0-mysql php7.0-ldap php7.0-zip php7.0-bcmath  php7.0-xml php7.0-common"
+pkg_dependencies="openssl"
+
+YNH_PHP_VERSION=7.3
+extra_php_dependencies=" php${YNH_PHP_VERSION}-mbstring  php${YNH_PHP_VERSION}-curl php${YNH_PHP_VERSION}-mysql php${YNH_PHP_VERSION}-ldap php${YNH_PHP_VERSION}-zip php${YNH_PHP_VERSION}-bcmath  php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-common"
```
Comment (maniackcrudelis) [inline scripts/_common.sh:11]: Here, the space at the beginning of the variable.

Comment (kay0u) [reply scripts/_common.sh]: Fixed

Comment (maniackcrudelis) [reply scripts/_common.sh]: Same error again...

Comment (maniackcrudelis) [reply scripts/_common.sh]: Try removing the double spaces maybe.

Comment (kay0u) [reply scripts/_common.sh]: Weird, it install without any problem on my `ynh-dev`, but yeah I missed one double space!

Comment (maniackcrudelis) [reply scripts/_common.sh]: It was on buster, trying on 3.8...

Comment (maniackcrudelis) [reply scripts/_common.sh]: Seems to work on 3.8, I upgrade my Buster VM.

**Comment by maniackcrudelis** [comment]
On 3.8, got that
```
Info:  [#############++.....] > Installing composer... [00h01m,31s]
Warning: yunohost.hook <lambda> - [1051.1] Your requirements could not be resolved to an installable set of packages.
Warning: yunohost.hook <lambda> - [1051.1] Unable to update core dependencies with Composer.
Warning: yunohost.hook <lambda> - [1051.1] Unable to update core dependencies with Composer.
```

File: scripts/_common.sh (hunk 1 above)
Comment (maniackcrudelis) [reply scripts/_common.sh]: Same error on Buster up to date
Probably not related directly to your code. I'll have a look to it later today.

Comment (maniackcrudelis) [reply scripts/_common.sh]: I did not reproduce directly on the VM, only into the lxc container.
So... I rebuild the container, I expect the upgrade didn't go well...

Comment (maniackcrudelis) [reply scripts/_common.sh]: Sorry my VM wasn't good...

**Comment by maniackcrudelis** [comment]
And again, my VM... it seems to actually work...

I rebuild the whole VM properly from an older snapshot, and I restart.

**Comment by maniackcrudelis** [comment]
Finally

- I confirm the error on 3.8
- Everything's ok on Buster.

**Comment by kay0u** [comment]
I'm able to install it on my ynh-dev buster:
https://paste.yunohost.org/tumumiwavu

**Comment by kay0u** [comment]
Now I'm able to install it on a fresh install! \o/

File: scripts/restore [hunk 2]
```diff
@@ -106,7 +106,7 @@ ynh_mysql_connect_as --user=$db_name --password=$db_pwd --database=$db_name < ./
 #=================================================
 ynh_script_progression --message="Reloading nginx web server and php-fpm..." --time --weight=1
 
-ynh_systemd_action --service_name=php7.0-fpm --action=reload
+ynh_systemd_action --service_name=php${YNH_PHP_VERSION}-fpm --action=reload
```
Comment (maniackcrudelis) [inline scripts/restore:109]: Situation we talked about.
On 3.8, you restart a service that doesn't exist.
php7.3 was never installed during your restore.

On nextcloud, we have managed that with a call the php helper
https://github.com/YunoHost-Apps/nextcloud_ynh/pull/292/files#diff-68321f788a23b70a33d17888e31bf3efR95-R104

**Comment by maniackcrudelis** [comment]
Except the restore script, everything was working on 3.8.
Still perfectly working on Buster.

File: scripts/restore (hunk 2 above)
Comment (kay0u) [reply scripts/restore:112]: Thank you for your tests and your feedback!

Should be fixed now

**Pull request merged**